File: week2/kuzey/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int currMin = prices[0];

        for (int each : prices) {
            if (each > currMin)
                max = Math.max(each - currMin, max);
            else
                currMin = Math.min(each, currMin);
        }

        return max;
    }
}

File: week2/omerSefaYilmaz/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let minPrice = prices[0];
    let maxProfit = 0;
    for (let i = 1; i < prices.length; i++) {
      maxProfit = Math.max(maxProfit, prices[i] - minPrice);
  
      if (prices[i] < minPrice) {
        minPrice = prices[i];
      }
    }
  
    return maxProfit;
  };